### DISCUSSION TITLE
Using Ray with Kirby CMS

### DISCUSSION THREAD

**Discussion by skyeewers** [spatie/ray#190] [Q&A] [answered]
Hey everyone!
I was wondering if there is any known way of using this package in the context of a [Kirby CMS](https://getkirby.com) project.  

Right now, if I go ahead and `composer require spatie/ray` inside the `public` folder of any minimal Kirby installation, I'll get an error whenever I try to load any page served by that Kirby instance.

The error complains about how smyfony's `dump.php` (part of ray's requirements) redeclares `dump()` as a function. This seems to be because Kirby, for some reason, also has a global `dump()` function as part of their global helpers.

Is there any reasonable way to work around this issue at the moment or is anyone trying to use ray with Kirby just out of luck for now?

Here's the full output of the error message mentioned above:

```
Fatal error: Cannot redeclare dump() (previously declared in /project_root/public/vendor/symfony/var-dumper/Resources/functions/dump.php:18) in /project_root/public/kirby/config/helpers.php on line 147
```

**Comment by freekmurze** [comment, marked as answer]
I don't think there's a solution for this. Ray uses Symfony's dump for several key functions, so we can't remove it. 

If possible, try to disable Kirby's `dump` function.

**Comment by afbora** [comment]
Kirby core functions can now be overridden. 

Ray plugin for Kirby CMS:
https://getkirby.com/plugins/genxbe/ray
